app/models/nomis/sentence_detail.rb

```diff
@@ -27,6 +30,20 @@ def conditional_release_date
       @conditional_release_override_date.presence || @conditional_release_date
     end
 
+    def recall_release_date
+      return decide_which_recall_release_date if @actual_parole_date.blank?
+      return @actual_parole_date if recall_release_override_date_is_earlier?
+      decide_which_recall_release_date
+    end
+  
+    def recall_release_override_date_is_earlier?
+      @post_recall_release_override_date.blank? || @post_recall_release_override_date > @actual_parole_date
+    end
+
+    def decide_which_recall_release_date
```

Comment:
I have a similar thought here:

I'm wondering whether this is a descriptive name of what the method gives us. It's true that this method does *decide* which value to return, but ultimately what the method gives us is the 'true' post recall release date (either the existing one, or the overridden one).

I'm wondering whether a more descriptive name could be something like `actual_post_recall_release_date`, or `true_post_recall_release_date` 🤔

Again: naming things is hard!